File: app/src/main/java/ru/sicampus/bootcamp2025/ui/login/LoginFragment.java
package com.example.myapplication.ui.login;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.text.Editable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.example.myapplication.ui.main.IventsActivity;
import com.example.myapplication.ui.main.MainActivity;
import com.example.myapplication.R;
import com.example.myapplication.core.SettingConstants;
import com.example.myapplication.utils.TextChangedListener;
import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;


public class LoginFragment extends Fragment {
    private static final String testEmail = "test@gmail.com";
    private static final String testPassword = "kloun";

    private TextInputEditText email;
    private TextInputEditText password;

    private TextInputLayout email_layout;
    private TextInputLayout password_layout;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_login, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        email = view.findViewById(R.id.email);
        password = view.findViewById(R.id.password);

        password_layout = view.findViewById(R.id.password_layout);
        email_layout = view.findViewById(R.id.email_layout);

        Button loginButton = view.findViewById(R.id.loginButton);
        TextView singupText = view.findViewById(R.id.singupText);


        email.addTextChangedListener(new TextChangedListener<>(email) {
            @Override
            public void onTextChanged(TextInputEditText target, Editable s) {
                listenerEmailEditText(target, s);
            }
        });

        password.addTextChangedListener(new TextChangedListener<>(password) {
            @Override
            public void onTextChanged(TextInputEditText target, Editable s) {
                listenerPasswordEditText(target, s);
            }
        });

        loginButton.setOnClickListener(this::onClickListenerLoginButton);
        //singupText.setOnClickListener(this::onClickListenerSingupText);
    }

    private void listenerEmailEditText(TextInputEditText target, Editable s) {
        if (s.toString().isEmpty()) {
            email_layout.setError("Обязательное поле");
        } else if (!isEmailValid(s.toString())) {
            email_layout.setError("Неверный формат");
        } else {
            email_layout.setErrorEnabled(false);
        }
    }

    private void listenerPasswordEditText(TextInputEditText target, Editable s) {
        if (s.toString().isEmpty()) {
            password_layout.setError("Обязательное поле");
        } else {
            password_layout.setErrorEnabled(false);
        }
    }

    private void onClickListenerLoginButton(View view) {
        // Отправка запроса на сервер
        // Получение и обработка ответа
        // Проверка ответа и переход на главную активность

        String emailString = String.valueOf(email.getText());
        String passwordString = String.valueOf(password.getText());
        SharedPreferences settings = view.getContext().getSharedPreferences(
                SettingConstants.PREFS_FILE, Context.MODE_PRIVATE
        );

        if (emailString.equals(testEmail) && passwordString.equals(testPassword)) {
            settings.edit().putString(SettingConstants.PREF_TOKEN, "token").apply();
            startActivity(new Intent(getActivity(), IventsActivity.class));
            requireActivity().finish();
        } else {
            Snackbar.make(view, "Некорректный email или пароль!", Snackbar.LENGTH_LONG).show();
        }
    }

   /* private void onClickListenerSingupText(View view) {
        requireActivity().getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragment_container_view, RegisterFragment.class, null)
                .commit();
    }*/

    private boolean isEmailValid(String email) {
        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }
}

File: app/src/main/java/ru/sicampus/bootcamp2025/ui/login/LoginFragment.kt
package ru.sicampus.bootcamp2025.ui.login

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import ru.sicampus.bootcamp2025.R
import ru.sicampus.bootcamp2025.databinding.FragmentLoginBinding
import java.io.IOException

class LoginFragment: Fragment(R.layout.fragment_login) {
    private var _binding: FragmentLoginBinding? = null
    private val binding: FragmentLoginBinding get() = _binding!!

    private val viewModel: LoginViewModel by viewModels()
    private lateinit var sharedPreferences: SharedPreferences
    private var savedLogin: String = ""

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        _binding = FragmentLoginBinding.bind(view)
        savedLogin = sharedPreferences.getString("LOGIN", "") ?: ""
        binding.login.setOnClickListener(this::onLoginButtonClicked)

    }

    private fun setupLoginButton() {
        binding.username.addTextChangedListener(object : TextWatcher{
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                binding.error.visibility = View.GONE
                val username = s.toString()
                binding.login.isEnabled = isUsernameValid(username)

            }

            override fun afterTextChanged(s: Editable?) {}
        })
    }

    private fun isUsernameValid(username: String): Boolean {
        val alf = "^[a-zA-Z0-9]+$".toRegex()
        return username.isNotEmpty() &&
                username.length >= 3 &&
                !username[0].isDigit() &&
                alf.matches(username)
    }
    override fun onAttach(context: Context) {
        super.onAttach(context)
        sharedPreferences = requireActivity().getSharedPreferences("LOGIN", Context.MODE_PRIVATE)
        setupLoginButton()
    }

    private fun onLoginButtonClicked(view: View) {
        val login = binding.username.text.toString()
        if (login.isEmpty()) return

        Thread {
            try {
                val authResult = viewModel.checkUserAuth(login)

                requireActivity().runOnUiThread {
                    if (authResult) processAuthSuccess(login) else processAuthError()
                }
            } catch (_: IOException) {
                processAuthError()
            }
        }.start()
    }

    private fun processAuthSuccess(login: String) {
        viewModel.saveUserLogin(login, sharedPreferences)
        findNavController().navigate(R.id.action_loginFragment_to_profileFragment)
    }

    private fun processAuthError() {
        binding.error.visibility = View.VISIBLE
    }
}